cmd/bundle/run.go

```diff
@@ -40,6 +43,9 @@ task or a Python wheel task, the second example applies.
 `,
 	}
 
+	var forcePull bool
+	cmd.Flags().BoolVar(&forcePull, "force-pull", false, "Skip local cache and load the state from the remote workspace")
```

Comment:
It seems to be a breaking change to skip state-pull by default. If your local bundle state is stale (doesn't have a deployed state for a particular job you want to run for example), then with this change the run will fail since we won't have an id to trigger it (while before this change the run will succeed).

Reply:
Yes indeed, this is a behavior change. I think it's important to make the command faster by default for interactive use cases. It's pretty unbearably slow for big workspaces right now.

Reply:
Why only for big workspaces? There shouldn't be anything in there that gets slower.

Reply:
You can achieve the same without changing the behavior by hoisting the following under the prompt:
```
			terraform.Interpolate(),
			terraform.Write(),
			terraform.StatePull(),
			terraform.Load(terraform.ErrorOnEmptyState),
```

The prompt only needs the job keys and job names, which we already know post-initialize.

Then to actually run it, we need the job ID, and therefore need to load the state.

Reply:
That would work for the non-prompt case. So we're not worried about changing behavior for the prompt case? (I am not since it helps so much in terms of performance)

Reply:
It doesn't change the behavior because loading the state is needed to get the resource IDs for the defined resources. To show the prompts, we only need the job keys (from the config) and the job names (we know them after the initialize phase).

Reply:
But with this proposed change, we'd still only pull after the prompt right? So we might not be showing the very latest state?

Reply:
That's correct. We only use the state file to get the ID. If someone has gone in and removed a job and then deployed it, the prompt will still show the name derived from the local configuration. That little amount of drift is acceptable, IMO, and I believe no different from what happens today.

Reply:
Updated to move state loading below the prompt.